packages/polling-controller/src/StaticIntervalPollingController.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseController, BaseControllerV1 } from '@metamask/base-controller';
+import type { NetworkClientId } from '@metamask/network-controller';
+import type { Json } from '@metamask/utils';
+
+import {
+  AbstractPollingControllerBaseMixin,
+  getKey,
+} from './AbstractPollingController';
+import type {
+  IPollingController,
+  PollingTokenSetId,
+} from './AbstractPollingController';
+
+type Constructor = new (...args: any[]) => object;
+
+/**
+ * StaticIntervalPollingControllerMixin
+ * A polling controller that polls on a static interval.
+ *
+ * @param Base - The base class to mix onto.
+ * @returns The composed class.
+ */
+function StaticIntervalPollingControllerMixin<TBase extends Constructor>(
+  Base: TBase,
+) {
+  abstract class StaticIntervalPollingController
+    extends AbstractPollingControllerBaseMixin(Base)
+    implements IPollingController
+  {
+    readonly #intervalIds: Record<PollingTokenSetId, NodeJS.Timeout> = {};
+
+    intervalLength: number | undefined = 1000;
+
+    setIntervalLength(intervalLength: number) {
+      this.intervalLength = intervalLength;
+    }
+
+    getIntervalLength() {
+      return this.intervalLength;
+    }
+
+    _startPollingByNetworkClientId(
+      networkClientId: NetworkClientId,
+      options: Json,
+    ) {
+      if (!this.intervalLength) {
+        throw new Error('intervalLength must be defined and greater than 0');
+      }
+
+      const key = getKey(networkClientId, options);
+      const existingInterval = this.#intervalIds[key];
+      this._stopPollingByPollingTokenSetId(key);
+
+      this.#intervalIds[key] = setTimeout(
+        async () => {
+          try {
+            await this._executePoll(networkClientId, options);
+          } catch (error) {
+            console.error(error);
+          }
+          this._startPollingByNetworkClientId(networkClientId, options);
+        },
+        existingInterval ? this.intervalLength : 0,
```

Comment:
Thoughts on using `getIntervalLength()` so that when you mock this method in the tests it'll work across the board?

Reply:
could do that... or just call `setIntervalLength` in the tests where you want o modify the default no?

Reply:
Yes, that would work too.

Reply:
done here: https://github.com/MetaMask/core/pull/3636/commits/823aaa188f572a8aa76973a012ae9bfa8da518e0